sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSComposeApp.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.joinforage.android.example.R
+import com.joinforage.android.example.ui.pos.screens.ActionSelectionScreen
+import com.joinforage.android.example.ui.pos.screens.MerchantSetupScreen
+
+enum class POSScreen(@StringRes val title: Int) {
+    MerchantSetupScreen(title = R.string.title_pos_merchant_setup),
+    ActionSelectionScreen(title = R.string.title_pos_action_selection)
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun POSComposeApp(
+    viewModel: POSViewModel = viewModel(),
+    navController: NavHostController = rememberNavController()
+) {
+    val backStackEntry by navController.currentBackStackEntryAsState()
+    val currentScreen = POSScreen.valueOf(
+        backStackEntry?.destination?.route ?: POSScreen.MerchantSetupScreen.name
+    )
+
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = { Text(stringResource(currentScreen.title)) },
+                navigationIcon = {
+                    if (navController.previousBackStackEntry != null) {
+                        IconButton(onClick = { navController.navigateUp() }) {
+                            Icon(
+                                imageVector = Icons.Filled.ArrowBack,
+                                contentDescription = stringResource(R.string.pos_back_button)
+                            )
+                        }
+                    }
+                }
+            )
+        }
+    ) { innerPadding ->
+        val uiState by viewModel.uiState.collectAsState()
+
+        NavHost(
+            navController = navController,
+            startDestination = POSScreen.MerchantSetupScreen.name,
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(innerPadding)
+                .padding(bottom = 72.dp, start = 16.dp, end = 16.dp)
+        ) {
+            composable(route = POSScreen.MerchantSetupScreen.name) {
+                MerchantSetupScreen(
+                    terminalId = "fake terminal ID",
```

Comment:
love the placeholder haha - I'm leaving this comment as a note to myself to actually read the Serial Number from the POS device's SDK to use instead of the place holder on POS devices.